rocketpy/motors/Motor.py

```diff
@@ -0,0 +1,1373 @@
+    def propellant_I_23(self):
+        """Inertia tensor 23 component of the propellant, the product of inertia
+        is relative to axes e_2 and e_3, centered at the instantaneous propellant
+        center of mass.
+
+        Parameters
+        ----------
+        t : float
+            Time in seconds.
+
+        Returns
+        -------
+        float
+            Propellant inertia tensor 23 component at time t.
+
+        Notes
+        -----
+        The e_2 direction is assumed to be the direction perpendicular to the
+        motor body axis, and perpendicular to e_1.
+        The e_3 direction is assumed to be the axial direction of the rocket
+        motor.
+        RocketPy follows the definition of the inertia tensor as in [1], which
+        includes the minus sign for all products of inertia.
+
+        References
+        ----------
+        .. [1] https://en.wikipedia.org/wiki/Moment_of_inertia
+        """
+        pass
+
+    @staticmethod
+    def reshapeThrustCurve(thrust, newBurnTime, totalImpulse):
+        """Transforms the thrust curve supplied by changing its total
+        burn time and/or its total impulse, without altering the
+        general shape of the curve.
+
+        Parameters
+        ----------
+        thrust : rocketpy.Function
+            Thrust curve to be reshaped.
+        newBurnTime : float, tuple of float
+            New desired burn time in seconds.
+        totalImpulse : float
+            New desired total impulse.
+
+        Returns
+        -------
+        rocketpy.Function
+            Reshaped thrust curve.
+        """
+        # Retrieve current thrust curve data points
+        timeArray, thrustArray = thrust.xArray, thrust.yArray
+        newBurnTime = tuple_handler(newBurnTime)
+
+        # Compute old thrust based on new time discretization
+        # Adjust scale
+        newTimeArray = (
+            (newBurnTime[1] - newBurnTime[0]) / (timeArray[-1] - timeArray[0])
+        ) * timeArray
+        # Adjust origin
+        newTimeArray = newTimeArray - newTimeArray[0] + newBurnTime[0]
+        source = np.column_stack((newTimeArray, thrustArray))
+        thrust = Function(
+            source, "Time (s)", "Thrust (N)", thrust.__interpolation__, "zero"
+        )
+
+        # Get old total impulse
+        oldTotalImpulse = thrust.integral(*newBurnTime)
+
+        # Compute new thrust values
+        newThrustArray = (totalImpulse / oldTotalImpulse) * thrustArray
+        source = np.column_stack((newTimeArray, newThrustArray))
+        thrust = Function(
+            source, "Time (s)", "Thrust (N)", thrust.__interpolation__, "zero"
+        )
+
+        return thrust
+
+    @staticmethod
+    def clipThrust(thrust, new_burn_time):
+        """Clips the thrust curve data points according to the new_burn_time
+        parameter. If the burn_time range does not coincides with the thrust
+        dataset, their values are interpolated.
+
+        Parameters
```

Comment:
Please add `new_burn_time` parameter here.
I don't want to guess what is it.

Reply:
Solved by 33c46c8b333d3fbde74423088d4f761000d66e69.